lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -19,3381 +19,16 @@
 
 namespace Doctrine\ORM\Mapping;
 
-use BadMethodCallException;
-use Doctrine\Instantiator\Instantiator;
-use InvalidArgumentException;
-use RuntimeException;
-use Doctrine\DBAL\Types\Type;
-use Doctrine\DBAL\Platforms\AbstractPlatform;
-use ReflectionClass;
-use Doctrine\Common\Persistence\Mapping\ClassMetadata;
-use Doctrine\Common\ClassLoader;
-use Doctrine\ORM\Cache\CacheException;
+@trigger_error(sprintf(
+    '%s is deprecated since 2.x and will be removed in 3.0. Use %s instead.',
```

Comment:
Note: Revisit message later, it may be ComponentMetadata instead of ClassMetadata.